Tests/Rules/UseDeclaredVarsMoreThanAssignments.tests.ps1

```diff
@@ -73,5 +73,10 @@ function MyFunc2() {
         It "returns no violations" {
             $noViolations.Count | Should -Be 0
         }
+
+        It "Using a variable via 'Get-Variable' does not trigger a warning" {
+            $noViolations = Invoke-ScriptAnalyzer -ScriptDefinition '$a=4; get-variable a' | Where-Object { $_.RuleName -eq $violationName }
```

Comment:
is the `where-object` needed?
could this just be 
```
Invoke-ScriptAnalyzer -ScriptDefinition '$a=4;Get-Variable a' | Should -BeNullOrEmpty
```

Reply:
The idea is to only assert against the rule under test being not fired for it to be more like a unit test. This allows for test isolation and less maintenance since there are already enough tests that do test the integration of rules.